syncstorage/storage/spanner.py

```diff
@@ -358,13 +371,68 @@ def _find_items(self, session, user, collection, **params):
                 bind_types[ts_var] = param_types.TIMESTAMP
 
         # Generate the query
+        fields = params.get("fields")
+        offset_params = params
+
+        if self._bsolm_index_separate and fields is None:
```

Comment:
It's not obvious why `and fields is None` here, is it to ensure we only apply this logic in `get_items` rather than `get_item_ids`?  If so, I wonder if it's worth refactoring `_find_items` into a couple of helper methods and pushing that logic up into `get_items`, like:

```
def get_items(...):
    if self._bsolm_index_separate:
        (params, ids) = self._do_bsolm_separate_query_to_get_the_ids()
    return self._find_items(**params)

def get_item_ids(...):
    return self._find_items(...)
```

Depends how much effort you want to invest in polishing this though, as opposed to just getting it out in stage to loadtest.

Reply:
That's right, to only apply it to get_items. I like the refactor suggestion but it's a little further complicated by encode_next_offset's need for the original params (`offset_params`).

I will keep it in mind if this is revisited. This is definitely a "get it out to stage and see" patch